samples/client.c

```diff
@@ -62,6 +62,8 @@ static char const *irq_to_str[] = {
     [VFU_DEV_REQ_IRQ] = "REQ"
 };
 
+static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Why do we need the mutex?

Reply:
The fake guest and the client use the same socket so we have to ensure the responses do not get mixed up.

Reply:
This problem should exist in the original version of the code as well, right? Do your changes make an existing bug easier to trigger? If so, I think it's simpler to have the fake guest not require a resposne since it only writes to BAR1.